src/librustc_driver/driver.rs

```diff
@@ -306,17 +307,10 @@ fn keep_hygiene_data(sess: &Session) -> bool {
     sess.opts.debugging_opts.keep_hygiene_data
 }
 
-
-/// The name used for source code that doesn't originate in a file
-/// (e.g. source from stdin or a string)
-pub fn anon_src() -> String {
-    "<anon>".to_string()
-}
-
-pub fn source_name(input: &Input) -> String {
+pub fn source_name(input: &Input) -> FileName {
     match *input {
         // FIXME (#9639): This needs to handle non-utf8 paths
```

Comment:
Is this FIXME still relevant in this place?

Reply:
I think it's fixed with this PR, but there's no repro example on the issue